### PR TITLE
Shaker_Sort in python issue #2155

### LINKED ISSUE #2155: Shaker Sort
- [x] C
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php
- [ ] Python - @chaithanyarlk
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by chaithanyarlk**
Fixes #2155

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Code File is added 
- 

#### Languages Used:
- Python 3
- 

#### Files Added:
- Shaker_Sort.py
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Shaker_Sort/Shaker_Sort.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#Python 3 implemnetation of Shaker_Sort
+
+def Shaker_Sort(arr):
+     n = len(arr)
```
Comment (jainaman224) [inline Shaker_Sort/Shaker_Sort.py]: You should use 4 space indentation. It is 5 space indented.

File: Shaker_Sort/Shaker_Sort.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     #getting the length of the listto get the end
```
Comment (jainaman224) [inline Shaker_Sort/Shaker_Sort.py]: Add space after `#`.

File: Shaker_Sort/Shaker_Sort.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#arr=[2, 10, 8, 1, 4, 1]
+#sample input this can be extended over user inputs even
+arr = list()
+arr = list(map(int,input("Enter array: ").strip().split(',')))
+Shaker_Sort(arr)
+print("After Shaker_Sort the list :")
+for i in range(0, len(arr), 1):
+     print(arr[i], end = " ")
+#outpu 1 1 2 4 8 10
```
Comment (jainaman224) [inline Shaker_Sort/Shaker_Sort.py]: Typo.

File: Shaker_Sort/Shaker_Sort.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#Python 3 implemnetation of Shaker_Sort
```
Comment (jainaman224) [inline Shaker_Sort/Shaker_Sort.py]: Typo

**Comment by chaithanyarlk** [comment]
Sorry for late reply I will correct them very soon

**Comment by chaithanyarlk** [comment]
I hope that I have made the required changes please review it and let me know any more changes required

File: Shaker_Sort/Shaker_Sort.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                # Left element is greater than that of right in list
+                arr[i], arr[i + 1] = arr[i + 1], arr[i]
+                # Swapping elements of the list
+                swapped = True
+        # Now the smallest element will be in it's correct place in the list
+        begin = begin + 1
+
+# Driver Code
+
+# Sample input arr = [2, 10, 8, 1, 4, 1]
```
Comment (somya-kapoor) [inline Shaker_Sort/Shaker_Sort.py]: add sample input and output at the end of the file

File: Shaker_Sort/Shaker_Sort.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+arr = list(map(int,input("Enter Array : ").stript().split(',')))
+Shaker_Sort(arr)
+print (" After Shaker_Sort the Array : ")
+for i in range (0, len(arr), 1) :
+    print (arr[i], end = " ")
+# Output 1 1 2 4 8 10
+
```
Comment (somya-kapoor) [inline Shaker_Sort/Shaker_Sort.py]: Remove extra line

**Comment by chaithanyarlk** [comment]
I have made the required changes please review it @jainaman224 ,  @somya-kapoor  and let me know if any modifications are required so that I can correct them as soon as possible

File: Shaker_Sort/Shaker_Sort.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        # Leave the lost element iterated from above loop as it is in correct place
+        for i in range (end - 1, begin - 1, - 1) :
+            if arr[i] > arr[i + 1] :
+                # Left element is greater than that of right in list
+                arr[i], arr[i + 1] = arr[i + 1], arr[i]
+                # Swapping elements of the list
+                swapped = True
+        # Now the smallest element will be in it's correct place in the list
+        begin = begin + 1
+# Driver Code
```
Comment (somya-kapoor) [inline Shaker_Sort/Shaker_Sort.py]: Leave a line before it

File: Shaker_Sort/Shaker_Sort.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+arr = list(map(int,input("Enter Array : ").stript().split(',')))
+Shaker_Sort(arr)
+# Call to Shaker_Sort function
+print (" After Shaker_Sort the Array : ")
+for i in range (0, len(arr), 1) :
+    print (arr[i], end = " ")
+# Sample input arr = [2, 10, 8, 1, 4, 1]
```
Comment (somya-kapoor) [inline Shaker_Sort/Shaker_Sort.py]: Leave a line before it

**Comment by chaithanyarlk** [comment]
I have made the required changes please review it @jainaman224 , @somya-kapoor and let me know if any modifications are required so that I can correct them as soon as possible

**Comment by somya-kapoor** [comment]
@chaithanyarlk Please squash the commits

**Comment by chaithanyarlk** [comment]
Can you @somya-kapoor  Please guide me over Squashing over commits as I have tried some online solutions but did not worked well .So I request to guide so that I will be able to finish this as soon as possible

**Comment by somya-kapoor** [comment]
@chaithanyarlk For reference http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

**Comment by somya-kapoor** [comment]
@chaithanyarlk Are you working on it?

**Comment by chaithanyarlk** [comment]
I have actually used GitHub Desktop to clone  and now I am facing problems .So I just want to reclone it again and I will update it . Is it okay?

**Pull request closed without merging**